src/workerd/io/actor-cache.c++

```diff
@@ -2446,6 +2535,18 @@ kj::Promise<void> ActorCache::startFlushTransaction() {
   // - All our cached entries are dirty.
   // - Dirty entries can only be removed from the cache map if replaced by a new dirty entry.
   // - Thus all new read requests for our cached entries keys will be served from cache.
+  // TODO(now) We actually want a separate forked promise chain for reads so that we don't block on
+  // write flushes.
+  if (getFlush.entries.size() == 1) {
+    auto entry = kj::mv(getFlush.entries.front());
+    getFlush.batches.clear();
+    getFlush.entries.clear();
+    co_await syncGet(storage, kj::mv(entry));
+  } else if (getFlush.entries.size() > 1) {
+    co_await syncGets(storage, kj::mv(getFlush));
```

Comment:
Does this structuring of the code -- where after doing our coalesced reads we have to go ahead and finish executing this `startFlushTransaction` method, including successfully flushing all of our writes to storage, before the `lastFlush` promise will resolve -- mean that our `storage.get()`/`storage.getMultiple()` promises no longer resolve until *after* any dirty writes have been flushed to disk?

If so, that feels like a meaningful latency regression for reads given that writes take much longer than reads.

If not, I'm not seeing what mechanism would cause the reads to resolve prior to this method finishing.

Reply:
> I think the most obvious (potential) issue here is that although reads can now resolve before the flush finishes, the next batch of reads will have to wait for the next flush

That isn't the end of the world, although I also don't see why it would be necessary (aside from it being more convenient in the code).